HPXMLtoOpenStudio/tests/test_validation.rb

```diff
@@ -0,0 +1,264 @@
+    errors = XMLHelper.validate(hpxml_doc.to_xml, File.join(schemas_dir, 'HPXML.xsd'), nil)
+    if errors.size > 0
+      puts "#{xml}: #{errors}"
+    end
+    assert_equal(0, errors.size)
+  end
+
+  def _get_hpxml_doc(target_xpath, mode)
+    @hpxml_docs.values.each do |hpxml_doc|
+      parent_element = target_xpath[0] == '' ? hpxml_doc : XMLHelper.get_element(hpxml_doc, target_xpath[0])
+      next if parent_element.nil?
+
+      child_elements = target_xpath[1]
+
+      if mode == 'deletion'
+        end_index = -1
+      elsif mode == 'addition'
+        end_index = -2
+      end
+
+      found_children = true
+      child_elements[0..end_index].each do |child_element|
+        next if XMLHelper.has_element(parent_element, child_element)
+
+        found_children = false
+        break
+      end
+      next unless found_children
+
+      return _deep_copy_object(hpxml_doc)
+    end
+
+    fail "Could not find HPXML file for target_xpath: #{target_xpath}."
+  end
+
+  def _get_element_name(assertion)
+    element_names = []
+    if assertion.partition(': ').last.start_with?('[not')
+      element_names << assertion.partition(': ').last.partition(' | ').last
+    else
+      element_name = assertion.partition(': ').last.partition(' | ').first
+      if element_name.count('[') != element_name.count(']')
+        diff = element_name.count('[') - element_name.count(']')
+        diff.times { element_name.concat(']') }
+      end
+      element_names << element_name
+
+      # Exceptions
+      # FIXME: Is there another way to handle this?
+      if assertion.partition(': ').last.include? 'AnnualHeatingDistributionSystemEfficiency'
+        # Need to remove both AnnualHeatingDistributionSystemEfficiency and AnnualCoolingDistributionSystemEfficiency
+        element_name_additional = assertion.partition(': ').last.partition(' | ').last
+        element_names << element_name_additional
+      elsif assertion.partition(': ').last.include? 'HousePressure'
+        # handle [(HousePressure and BuildingAirLeakage/UnitofMeasure[text()!="ACHnatural"]) or (not(HousePressure) and BuildingAirLeakage/UnitofMeasure[text()="ACHnatural"])]
+        element_names[0] = 'HousePressure' # replacing element name with 'HousePressure' for the test (i.e. the test by element deletion).
+      end
+    end
+
+    return element_names
+  end
+
+  def _get_expected_error_message(parent_xpath, assertion)
+    if parent_xpath == '' # root element
+      return [[assertion.partition(': ').first, parent_xpath].join(': '), assertion.partition(': ').last].join() # return "Expected x element(s) for xpath: foo"
+    else
+      return [[assertion.partition(': ').first, parent_xpath].join(': '), assertion.partition(': ').last].join(': ') # return "Expected x element(s) for xpath: foo: bar"
+    end
+  end
+
+  def _get_child_element_with_value(target_xpath, max_number_of_elements_allowed)
```

Comment:
The method is called "_get_child_element_with_value" but it returns an array. Please update the method name and provide a brief comment describing what the method does.